FinniversKit/Sources/Components/Infobox/InfoboxSwiftUIView.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import SwiftUI
+
+public struct InfoboxSwiftUIView: View {
+    public let style: InfoboxView.Style
+    public var viewModel: InfoboxViewModel?
+    var onPrimaryButtonTapped: (() -> Void)?
```

Comment:
Since this view uses a view model I like to put view behaviour and "business" logic inside the view model. You can then easily reference the handler like `Button(action: viewModel.onPrimaryButtonTapped)` since the signature is the same. The view model can also be easily tested. The style can go in the view model too.

Reply:
This is just a suggestion though, the "what goes where" debate is endless. 😄

Reply:
I don't disagree 😄  

However...

I think the reason I haven't done so already is that I've reused the `InfoboxViewModel` that already exists and is used with the existing `UIKIt` `InfoboxView`. Here the `InfoboxView` has:

- a `InfoboxViewModel`
- a `style` (`InfoboxView.Style`)
- a `delegate` (`InfoboxViewDelegate`)

And we can discuss if that is a pretty solution or not...it is what it is 🤷 

After reading your suggestion I tried adding `var onPrimaryButtonTapped: (() -> Void)? { get set }` to the `InfoboxViewModel` but that is a protocol so that just opened a new can of worms and would require changes to the existing implementations of `InfoboxViewModel` so....I slowly backed away 😄